core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/UnwrapCastInComparison.java

```diff
@@ -171,6 +172,14 @@ public Expression rewriteComparisonExpression(ComparisonExpression node, Void co
             return unwrapCast(expression);
         }
 
+        @Override
+        public Expression rewriteBetweenPredicate(BetweenPredicate node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+        {
+            return and(
```

Comment:
you want to call `treeRewriter.defaultRewrite` to ensure between's child nodes are processed

you want to call `unwrapCast` to do something meaningful. Single pass of UnwrapCastInComparison.Visitor should do the job of unwrapping casts, and not require another pass to do its job right